src/components/AssetDetailComponent.jsx

```diff
@@ -75,25 +69,27 @@ export class AssetDetailComponent extends Component {
       <NavbarComponent>
         <Container>
           <Header as="h1" content="Asset Detail" className="asset-detail-header" />
-          <AssetDetailContent
-            {...this.props}
-            assetDetail={this.props.assetDetail}
-            assignedUser={this.state.assignedUser}
-            errorMessage={this.state.errorMessage}
-            hasError={this.state.hasError}
-            isLoading={this.props.isLoading}
-            onSelectUserEmail={this.onSelectUserEmail}
-            handleAssign={this.handleAssign}
-            handleUnassign={this.handleUnassign}
-            selectedUserId={this.state.selectedUser}
-            show={this.show}
-            handleConfirm={this.handleConfirm}
-            handleCancel={this.handleCancel}
-            buttonState={this.props.buttonLoading}
-            assignAssetButtonState={this.state.assignAssetButtonState}
-            selectedUser={this.state.selectedUser}
-            users={this.props.assetAsigneeUsers}
-          />
+          { isEmpty(this.props.assetDetail) ?
+            <LoaderComponent loadingText="Loading" /> :
+            <AssetDetailContent
+              {...this.props}
+              assetDetail={this.props.assetDetail}
+              assignedUser={this.state.assignedUser}
+              errorMessage={this.state.errorMessage}
+              hasError={this.state.hasError}
+              isLoading={this.props.isLoading}
+              onSelectUserEmail={this.onSelectUserEmail}
+              handleAssign={this.handleAssign}
```

Comment:
What happened to moving methods such as this to the `AssetDetailContent` component? This component takes in more props that it can contain in its own class